homeassistant/components/samsungtv/bridge.py

```diff
@@ -375,13 +378,18 @@ def _try_connect(self) -> str:
 
     async def async_device_info(self) -> dict[str, Any] | None:
         """Try to gather infos of this TV."""
-        return await self.hass.async_add_executor_job(self._device_info)
+        if self.port:
```

Comment:
```suggestion
        if not self.port:
            return None
```

Then outdent below

Reply:
But that influences coverage :(

Reply:
The branching wasn't covered in the first place so it would be good to cover it.  The CI is only reporting line coverage.

Reply:
I originally added the condition due to the failure you communicated before:
```
 File "/Users/bdraco/home-assistant/homeassistant/components/samsungtv/config_flow.py", line 211, in async_step_user
    await self._async_set_device_unique_id(raise_on_progress=False)
  File "/Users/bdraco/home-assistant/homeassistant/components/samsungtv/config_flow.py", line 104, in _async_set_device_unique_id
    if not await self._async_get_and_check_device_info():
  File "/Users/bdraco/home-assistant/homeassistant/components/samsungtv/config_flow.py", line 141, in _async_get_and_check_device_info
    _port, _method, info = await async_get_device_info(
  File "/Users/bdraco/home-assistant/homeassistant/components/samsungtv/bridge.py", line 65, in async_get_device_info
    return bridge.port, bridge.method, await bridge.async_device_info()
  File "/Users/bdraco/home-assistant/homeassistant/components/samsungtv/bridge.py", line 391, in async_device_info
    device_info: dict[str, Any] = await self._rest_api.rest_device_info()
  File "/Users/bdraco/home-assistant/venv/lib/python3.9/site-packages/samsungtvws/async_rest.py", line 70, in rest_device_info
    return await self._rest_request("")
  File "/Users/bdraco/home-assistant/venv/lib/python3.9/site-packages/samsungtvws/async_rest.py", line 61, in rest_request
    async with future as resp:
  File "/Users/bdraco/home-assistant/venv/lib/python3.9/site-packages/aiohttp/client.py", line 1138, in __aenter__
    self._resp = await self._coro
  File "/Users/bdraco/home-assistant/venv/lib/python3.9/site-packages/aiohttp/client.py", line 424, in _request
    raise InvalidURL(str_or_url) from e
 aiohttp.client_exceptions.InvalidURL: http://192.168.107.78:None/api/v2/
```

However, I can't seem to replicate the issue, and in all my tests the port gets defined inside the try_connect routine.

Are you able to replicate the original error if you strip those two lines ?
```python
    if not self.port:
        return None
```